build/root/Makefile.generated_files

```diff
@@ -408,81 +408,97 @@ $(CONVERSION_GEN): $(k8s.io/kubernetes/vendor/k8s.io/code-generator/cmd/conversi
 # The result file, in each pkg, of open-api generation.
 OPENAPI_BASENAME := $(GENERATED_FILE_PREFIX)openapi
 OPENAPI_FILENAME := $(OPENAPI_BASENAME).go
-OPENAPI_OUTPUT_PKG := pkg/generated/openapi
-CRD_OPENAPI_OUTPUT_PKG := staging/src/k8s.io/apiextensions-apiserver/pkg/generated/openapi
 BOILERPLATE_FILENAME := vendor/k8s.io/code-generator/hack/boilerplate.go.txt
 REPORT_FILENAME := $(OUT_DIR)/violations.report
 IGNORED_REPORT_FILENAME := $(OUT_DIR)/ignored_violations.report
-KNOWN_VIOLATION_FILENAME := api/api-rules/violation_exceptions.list
+
 # When UPDATE_API_KNOWN_VIOLATIONS is set to be true, let the generator to write
 # updated API violations to the known API violation exceptions list.
 ifeq ($(UPDATE_API_KNOWN_VIOLATIONS),true)
-    REPORT_FILENAME:=$(KNOWN_VIOLATION_FILENAME)
+    REPORT_FILENAME:=$(KUBE_KNOWN_VIOLATION_FILENAME)
```

Comment:
Fixed.

This code path was even broken before this PR because we forgot the dependency from the violation file (which is touched) to the output. I fixed that. And I added a test case.